Update RELEASE.md (#3101)
greenbone/gsa: RELEASE.md

Before creating a new release please do a careful consideration about the
version number for the new release. We are following
[Calendar Versioning](https://calver.org/) with GSA 20.08.

* Fetch upstream changes and create release branch

  ```sh
  git fetch upstream

>   git checkout -b <username>/prepare-new-release upstream/gsa-21.04

Reviewer: ```suggestion
  git checkout -b <username>/prepare-new-release upstream/gsa-<version>
```

* Update [CHANGELOG.md](https://github.com/greenbone/gsa/blob/master/CHANGELOG.md)

> * Set the `dev` version to `0` in these files:
> `set (PROJECT_DEV_VERSION 1)` -> `set (PROJECT_DEV_VERSION 0)`
> [CMakeLists.txt](https://github.com/greenbone/gsa/blob/master/CMakeLists.txt), [gsa/CMakeLists.txt](https://github.com/greenbone/gsa/blob/master/gsa/CMakeLists.txt), and [gsad/CMakeLists.txt](https://github.com/greenbone/gsa/blob/master/gsad/CMakeLists.txt)

Reviewer: ```suggestion
* Unset the `dev` version to `0` in [CMakeLists.txt](https://github.com/greenbone/gsa/blob/master/CMakeLists.txt), [gsa/CMakeLists.txt](https://github.com/greenbone/gsa/blob/master/gsa/CMakeLists.txt), and [gsad/CMakeLists.txt](https://github.com/greenbone/gsa/blob/master/gsad/CMakeLists.txt):
`set (PROJECT_DEV_VERSION 1)` -> `set (PROJECT_DEV_VERSION 0)`
* Remove the `.dev1` from the version string in [package.json](https://github.com/greenbone/gsa/blob/master/gsa/package.json) and [version.js](https://github.com/greenbone/gsa/blob/master/gsa/src/version.js)
```

* Update [CHANGELOG.md](https://github.com/greenbone/gsa/blob/master/CHANGELOG.md)

> * Update version in [CMakeLists.txt](https://github.com/greenbone/gsa/blob/master/CMakeLists.txt), [CMakeLists.txt](https://github.com/greenbone/gsa/blob/master/gsa/CMakeLists.txt), and [CMakeLists.txt](https://github.com/greenbone/gsa/blob/master/gsad/CMakeLists.txt)
> 

Reviewer: ```suggestion
* Update version in [CMakeLists.txt](https://github.com/greenbone/gsa/blob/master/CMakeLists.txt), [CMakeLists.txt](https://github.com/greenbone/gsa/blob/master/gsa/CMakeLists.txt), and [CMakeLists.txt](https://github.com/greenbone/gsa/blob/master/gsad/CMakeLists.txt)
  * It should be enough to set the `dev` version to `0` in these files: 
  `set (PROJECT_DEV_VERSION 1)` -> `set (PROJECT_DEV_VERSION 0)`
```

  * Make sure that the source code tarballs contain the same data as the git repository
  * Sign _all four_ files with your key
  ```sh
  gpg --default-key 0ED1E580 --detach-sign -a xyz.tar.gz
  gpg --default-key 0ED1E580 --detach-sign -a xyz.zip
  ```
  This will create files with an .asc file type. If .sig files were created, you probably didn't use the option `--armor` or `-a`.
  * Rename the node modules files and their signatures with the approriate version into, e.g. `gsa-node-modules-<version>.tar.gz`.
  * Edit your release again and add the following files as binary
    * the .tar.gz and .zip files of the node modules
    * both of their detached signatures
    * both detached signatures of the source code tarballs
  * Click "Update release"

  For further information see https://help.github.com/articles/creating-releases/

  ## Cleanup after the release
  * Create a new, empty unreleased section in the `CHANGELOG.md` file
  * Update the version in _all_ version files:

>     * increment the version in the `CMakeLists.txt` files, and set `dev` version again

Reviewer: ```suggestion
    * increment the version in the `CMakeLists.txt` files, and set `dev` version again (`set (PROJECT_DEV_VERSION 1)`)
```

>     * increment the version in [package.json](https://github.com/greenbone/gsa/blob/gsa-21.04/gsa/package.json) and [version.js](https://github.com/greenbone/gsa/blob/gsa-21.04/gsa/src/version.js).

Reviewer: ```suggestion
    * increment the version in [package.json](https://github.com/greenbone/gsa/blob/master/gsa/package.json) and [version.js](https://github.com/greenbone/gsa/blob/master/gsa/src/version.js).
```

  * Go to the "Releases" section on GitHub and download the source code tarballs created automatically by GitHub
  * Go to your source directory in your development environment and compress the `node_modules` as tar.gz and .zip. If you change branches a lot, before you compress the directory, you might want to make sure that the node modules are actually correct by deleting the folder and creating it again by running `yarn`.
  * Make sure that the source code tarballs contain the same data as the git repository
  * Sign _all four_ files with your key
  ```sh
  gpg --default-key 0ED1E580 --detach-sign -a xyz.tar.gz
  gpg --default-key 0ED1E580 --detach-sign -a xyz.zip
  ```
  This will create files with an .asc file type. If .sig files were created, you probably didn't use the option `--armor` or `-a`.
  * Rename the node modules files and their signatures with the approriate version into, e.g. `gsa-node-modules-<version>.tar.gz`.
  * Edit your release again and add the following files as binary
    * the .tar.gz and .zip files of the node modules
    * both of their detached signatures
    * both detached signatures of the source code tarballs
  * Click "Update release"

> 

Reviewer: ```suggestion

## Cleanup after the release

* Create a new, empty unreleased section in the `CHANGELOG.md` file
* Update the version in _all_ version files:
  * increment the version in the `CMakeLists.txt` files, and set `dev` version again
  * increment the version in `gsa/src/version.js` and `gsa/package.json`

```

Author: Incrementing the versions here again will make the previous steps with these files incorrect.

Reviewer: Yeah but this is the correct place to increment the versions.
Technically the versions should be incremented here, with a `dev` version. And Before the release only the `dev` should be removed.

Reviewer: Yeah I would remove it in the top of the page and increment and set dev *after*  release. 
Than only remove the dev thingies before release.

The About page line is also obsolete now, since we use the `gsa/src/version.js` file now.

Author: That line is not in the PR. Are you sure you are reviewing the correct version?
